core/server/common/src/main/java/alluxio/util/FeatureUtils.java

```diff
@@ -35,7 +35,11 @@ public static boolean isEmbeddedJournal() {
    * @return true, if running with rocks
    */
   public static boolean isRocks() {
-    return Configuration.get(PropertyKey.MASTER_METASTORE) == MetastoreType.ROCKS;
+    if (Configuration.isSetByUser(PropertyKey.MASTER_METASTORE)) {
+      return Configuration.get(PropertyKey.MASTER_METASTORE) == MetastoreType.ROCKS;
+    }
+    return Configuration.get(PropertyKey.MASTER_INODE_METASTORE) == MetastoreType.ROCKS
+        &&  Configuration.get(PropertyKey.MASTER_BLOCK_METASTORE) == MetastoreType.ROCKS;
```

Comment:
This part is actually quite interesting. Does "Inode-on-rocks + block-on-heap" count as rocks or heap? @elega @dbw9580 WDYT?

The good news is this check is only used in `UpdateCheck` which is a minor feature in Alluxio.

Reply:
https://github.com/Alluxio/alluxio/blob/master/core/server/master/src/main/java/alluxio/master/meta/UpdateCheck.java#L124-L136
Looking at how this method is used, I feel like the semantics of "isRocks" is closer to "isRocksUsed". The result of the method will be attached in the rpc header as a user agent for update check. So I think it's an OR condition instead of AND.

Reply:
How about separate the method to isInodeRocks() and isBlockRocks()?